app/src/main/java/io/gnosis/safe/ui/safe/send_funds/SendAssetReviewViewModel.kt

```diff
@@ -0,0 +1,102 @@
+package io.gnosis.safe.ui.safe.send_funds
+
+import io.gnosis.data.models.Safe
+import io.gnosis.data.repositories.SafeRepository
+import io.gnosis.data.repositories.TransactionRepository
+import io.gnosis.data.utils.SemVer
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import pm.gnosis.model.Solidity
+import java.math.BigInteger
+import javax.inject.Inject
+
+class SendAssetReviewViewModel
+@Inject constructor(
+    private val transactionRepository: TransactionRepository,
+    private val safeRepository: SafeRepository,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<SendAssetReviewState>(appDispatchers) {
+
+    lateinit var activeSafe: Safe
+        private set
+
+    private var safeNonce: BigInteger? = null
+    private var minSafeNonce: BigInteger? = null
+    private var safeTxGas: BigInteger? = null
+    private var proposedSafeTxGas: BigInteger? = null
+
+    override fun initialState(): SendAssetReviewState =
+        SendAssetReviewState(viewAction = null)
+
+    init {
+        safeLaunch {
+            activeSafe = safeRepository.getActiveSafe()!!
+        }
+    }
+
+    fun loadTxEstimationData(
+        chainId: BigInteger,
+        from: Solidity.Address,
+        to: Solidity.Address,
+        value: BigInteger
+    ) {
+        safeLaunch {
+            val txEstimation = transactionRepository.estimateTransaction(
+                chainId,
+                from,
+                to,
+                value
+            )
+            minSafeNonce = txEstimation.currentNonce
+            if (safeNonce == null) {
+                safeNonce = txEstimation.recommendedNonce
+            }
+            if (SemVer.parse(activeSafe.version!!, ignoreExtensions = true) < SemVer(1, 3, 0)) {
+                proposedSafeTxGas = txEstimation.safeTxGas
+                if (safeTxGas == null) {
+                    safeTxGas = proposedSafeTxGas
+                }
+            }
+        }
+    }
+
+    fun onAdvancedParamsEdit() {
+        safeLaunch {
+            updateState {
+                SendAssetReviewState(
+                    viewAction = ViewAction.NavigateTo(
+                        SendAssetReviewFragmentDirections.actionSendAssetReviewFragmentToEditAdvancedParamsFragment(
+                            activeSafe.chain,
+                            safeNonce.toString(),
+                            minSafeNonce.toString(),
+                            safeTxGas.toString(),
```

Comment:
null is converted to "null" here and crashes in EditAdvancedParamsFragment with a numberFormatException. Maybe convert null to "0"?

Reply:
what exactly was null here? safeNonce and minSafeNonce are required, but safeGas params can be null and EditAdvancedParamsFragment has null checks like

```
private val safeTxGas by lazy { navArgs.safeTxGas?.let { BigInteger(it) } }
```

Do you have steps to reproduce your crash?

Reply:
Same for `safeNonce` and `proposedSafeTxGas`

Reply:
The following would work. But maybe there is a better way to ennsure they are not null here:

```
SendAssetReviewFragmentDirections.actionSendAssetReviewFragmentToEditAdvancedParamsFragment(
                            activeSafe.chain,
                            safeNonce?.let { safeNonce.toString() } ?: "0",
                            minSafeNonce.toString(),
                            safeTxGas?.let { safeTxGas.toString() } ?:  "0",
                            proposedSafeTxGas?.let { proposedSafeTxGas.toString()} ?: "0"
                        )
```

Reply:
`it` is  a String there. The String  "null"

Reply:
solved 👍